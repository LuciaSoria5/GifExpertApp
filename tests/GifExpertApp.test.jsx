const { render, screen, fireEvent } = require("@testing-library/react");
const { GifExpertApp } = require("../src/GifExpertApp");


describe('Prueba en <GifExpertApp />', () => { 
    
    test('Debe matchear con el snapchot', () => { 
        const { container } = render(<GifExpertApp />);
        expect( container ).toMatchSnapshot();
     });

     test('Se debe agregar una nueva categoria', () => { 
        const { container } = render(<GifExpertApp />);
        const categoria = 'Naruto';

        const input = screen.getByRole('textbox');
        const form = screen.getByRole( 'form' );

        fireEvent.input(input, { target: { value: categoria } });
        fireEvent.submit(form);

        expect( screen.getAllByText(categoria).length ).toBe( 1 );
      });

      test('No se deben agregar categorias repetidas', () => { 
        const { container } = render(<GifExpertApp />);
        const categoria = 'Spy x Family';
        
        const input = screen.getByRole('textbox');
        const form = screen.getByRole( 'form' );

        fireEvent.input(input, { target: { value: categoria } });
        fireEvent.submit(form);

        expect( screen.getAllByText(categoria).length ).toBe( 1 );
      });
 });