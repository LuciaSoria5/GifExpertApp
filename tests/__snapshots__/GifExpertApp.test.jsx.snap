// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifExpertApp /> Debe matchear con el snapchot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value="Spy x Family"
    />
  </form>
  <h3>
    Spy x Family
  </h3>
  <h2>
    Cargando im√°genes...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
